### PR TITLE
Add functionality

### PR DESCRIPTION
**Pull request by engelca**
Adding in the ability to replace land/surface from an existing start dump file, and correcting paths used when using the BARRA-R2 archive.

### CODE REVIEW

File: src/replace_landsurface_with_FF_IC.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import iris
+import mule
+import os
+import sys
+import numpy as np
+from glob import glob
+from pathlib import Path
+import xarray as xr, sys, argparse
+from datetime import datetime,timedelta
```
Comment (CodeGat) [inline src/replace_landsurface_with_FF_IC.py:16]:
```suggestion
import mule
```
Just suggesting the removal of the unused imports here :)

File: src/replace_landsurface_with_FF_IC.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    Function to get the land/surface data from another fields file into the start dump.
+
+    Parameters
+    ----------
+    mask_fullpath : Path
+        Path to the mask defining the spatial extent
+    ic_file_fullpath : string
+        Path to file with the coarser resolution data to be replaced with ".tmp" appended at end
+    ic_date : string
+        The date-time required in "%Y%m%d%H%M" format
```
Comment (CodeGat) [inline src/replace_landsurface_with_FF_IC.py:45]: `source_fullpath` will need an entry in this parameter list

File: src/hres_eccb.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 
 import replace_landsurface_with_ERA5land_IC
 import replace_landsurface_with_BARRA2R_IC
+import replace_landsurface_with_FF_IC
```
Comment (CodeGat) [inline src/hres_eccb.py:24]: This import isn't used in this file, I don't think.

**Review by CodeGat — COMMENTED** [review]
Just a couple of small things. Otherwise looks good to me :)

File: src/hres_eccb.py (hunk 3 above)
Comment (engelca) [reply src/hres_eccb.py:24]: Yes you are correct Tommy.

File: src/replace_landsurface_with_FF_IC.py (hunk 2 above)
Comment (engelca) [reply src/replace_landsurface_with_FF_IC.py:45]: Yes, I will add that in thanks Tommy.

**Comment by atteggiani** [comment]
Thank you for the review comments @CodeGat.

I marked this PR as draft for now, because its head branch [add-functionality](https://github.com/ACCESS-NRI/replace_landsurface/tree/add-functionality) has a few missing commits.

The new branch [add-functionality_corrected](https://github.com/ACCESS-NRI/replace_landsurface/tree/add-functionality_corrected) has been created and will likely be used instead. 

@engelca please consider adding Tommy's comments in the `add-functionality_corrected` branch instead of this one. 

Thank you.

**Comment by atteggiani** [comment]
Closed as substituted by an [updated PR](#33)

**Pull request closed without merging**